src/encoder/stateless/h264/predictor.rs

```diff
@@ -101,15 +101,15 @@ impl<Picture, Reference> LowDelayH264<Picture, Reference> {
             .timing_info(1, self.tunings.framerate * 2, false)
             .build();
 
-        const MIN_QP: u8 = 1;
-        const MAX_QP: u8 = 51;
+        let min_qp = self.tunings.min_quality.max(MIN_QP as u32);
+        let max_qp = self.tunings.max_quality.min(MAX_QP as u32);
 
         let init_qp = if let RateControl::ConstantQuality(init_qp) = self.tunings.rate_control {
             // Limit QP to valid values
-            init_qp.clamp(MIN_QP as u32, MAX_QP as u32) as u8
+            init_qp.clamp(min_qp, max_qp) as u8
```

Comment:
General question: I'm wondering if we couldn't/shouldn't turn all these clamped values into their own types that implement `From<u32>` and enforce that the value is clamped. Then the use of the type would ensure we haven't accidentally missed clamping a value somewhere.

Reply:
I do like this idea and I'd love to do that. However I've decided to create a `Tunings`  struct codec agnostic. I'd like to avoid creating an enum with `StatelessVideoEncoder<Handle, Codec>` (in future `VideoEncoder<...>`) per each codec in our use cases, which would be necessary to accommodate those types in `Tunings`.